### PR TITLE
Handled offline flash

### PR DESCRIPTION
**Pull request by vojtatranta**

### CODE REVIEW

File: wzk/net/FlashMiddleware.coffee [hunk 1]
```diff
@@ -59,5 +60,11 @@ class wzk.net.FlashMiddleware
     @flash.addChild msg
     [msg]
 
+  offline: ->
+    msg = new wzk.ui.FlashMessage dom: @flash.getDomHelper(), msg: @msgs['offline'], severity: 'info', fadeOut: true, closable: true
```
Comment (LukasRychtecky) [inline wzk/net/FlashMiddleware.coffee]: Better to create a FlashMessage instance from method (duplicated code)

File: wzk/net/XhrFactory.coffee [hunk 2]
```diff
@@ -60,7 +75,37 @@ class wzk.net.XhrFactory
...
+    @protected
+    @param {wzk.net.XhrIo} xhr
+    @param {wzk.net.FlashMiddleware} flash
+  ###
+  handleCriticalError: (xhr, flash) ->
+    return if not flash
+    isOnline = @isOnline()
+    if xhr.getStatus() is 0 and isOnline is false and @wasOnline()
+      @flash.offline()
+    else if xhr.getStatus() is 500
```
Comment (LukasRychtecky) [inline wzk/net/XhrFactory.coffee]: better check status >= 500

File: wzk/net/XhrFactory.coffee [hunk 3]
```diff
@@ -60,7 +75,37 @@ class wzk.net.XhrFactory
...
+      @flash.error xhr.getStatus()
```
Comment (LukasRychtecky) [inline wzk/net/XhrFactory.coffee]: You should display a human friely message.

Comment (vojtatranta) [inline wzk/net/XhrFactory.coffee]: Messages translation for messages must be added to base template.

Offline by default says Offline, flash.error says Interní chyba, prosím, opakujte později.

File: wzk/net/FlashMiddleware.coffee [hunk 4]
```diff
@@ -54,10 +55,22 @@ class wzk.net.FlashMiddleware
...
+  offline: ->
+    [@createInfoFlash @msgs['offline'], 'info']
+
+  ###*
+    @param {string} msg
+    @param {string=} cls
+    @param {boolean=} closable
+    @return {wzk.ui.FlashMessage}
+  ###
+  createInfoFlash: (msg, cls = null, closable = true) ->
```
Comment (LukasRychtecky) [inline wzk/net/FlashMiddleware.coffee]: createFlash insteadof createFlashInfo

Comment (vojtatranta) [inline wzk/net/FlashMiddleware.coffee]: But it creates online info flash

Comment (LukasRychtecky) [inline wzk/net/FlashMiddleware.coffee]: Sorry

**Pull request merged**